### DISCUSSION TITLE
Plot using Y1 and Y2

### DISCUSSION THREAD

**Discussion by jwolff-ncar** [dtcenter/METplus#1123] [Plot Generation] [answered]
Hi,
I am trying to plot two variables (UGRD and VGRD) on one plot for BCRMSE and ME. 

I have a plot/xml for U [plot_U_BCRMSE_ME.xml.txt](https://github.com/dtcenter/METplus/files/7079824/plot_U_BCRMSE_ME.xml.txt) and V [plot_V_BCRMSE_ME.xml.txt](https://github.com/dtcenter/METplus/files/7079823/plot_V_BCRMSE_ME.xml.txt) separately. When I try to put U and V on the same y1 axis I get an error "Failed to create a plot. fcst_var must remain constant for MODE, MTD, Aggregation statistics, Event Equalizer". So I tried to put V on y2 and leave U on y1 but when I do that only V shows up [plot_UV_BCRMSE_ME.xml.txt](https://github.com/dtcenter/METplus/files/7079825/plot_UV_BCRMSE_ME.xml.txt). I am probably setting something wrong somewhere but any help you can provide would be very appreciated.
Thanks!

**Comment by TatianaBurek** [comment]
@jwolff-ncar Jamie, I believe that  you should be able to make this plot and the bug in METviewer prevents you to do that.
I created bug issues in METcaclpy (for Python) and METviewer(Rscript) and assigned them to beta3:
https://github.com/dtcenter/METcalcpy/issues/113
https://github.com/dtcenter/METviewer/issues/328

**Reply by jwolff-ncar** [reply]
Thank you for taking a look at this issue @TatianaBurek. I look forward to the bug fix as it becomes available.

**Comment by TatianaBurek** [comment, marked as answer]
@jwolff-ncar Jamie,
I fixed the code and it is possible now to plot different variables on y1 and y2 axis. The fix is deployed on dakota http://dakota.rap.ucar.edu:8080/metviewer_dev/
and will be the part of beta3 release.
Please note, that if you use Rscript and sync axis you might not see y2 series because the sync happens based on y1 axis.  You need to explicitly provide Y1 Bounds to fix this. Python doesn't have this problem

**Reply by JohnHalleyGotway** [reply]
@jwolff-ncar once you've tested and confirmed the fix, please feel free to select one of the messages on this thread as being the "Answer".

**Comment by jwolff-ncar** [comment]
@TatianaBurek I uploaded the plot_UV_BCRMSE_ME.xml to the dakota instance but it still did not create a plot with U/V together. I do not have the Y1/Y2 axes synced so I don't think that is the issue. Could you send me the xml that you got to work and I will see how it differs from what I have set?  Thanks!

**Comment by TatianaBurek** [comment]
@jwolff-ncar  plot_UV_BCRMSE_ME.xml should work on dakota : http://dakota.rap.ucar.edu:8080/metviewer_dev/
Make sure to remove 'Job title' or check 'Keep Revisions'. Otherwise you could get older versions of the image.
Please, test with Rscript and Python

**Reply by jwolff-ncar** [reply]
I was able to get this to work today and confirm it looks good. Thanks!

**Reply by JohnHalleyGotway** [reply]
@jwolff-ncar thanks for confirming the fix. I'll go ahead and lock this discussion. If more issues or questions arise, please feel free to post a new discussion.
